cmd/soroban-cli/src/commands/contract/mod.rs

```diff
@@ -40,6 +41,9 @@ pub enum Cmd {
     #[command(subcommand)]
     Id(id::Cmd),
 
+    /// Initialize a Soroban project with an example contract
+    Init(init::Cmd),
```

Comment:
I think we want this to be a top-level command, right? If we nest it under `contract`, it makes it seem like it is for initializing a contract. But it's more general than that.

Reply:
@leighmcculloch suggested we move it under `contract` as we are starting to think about merging soroban-cli and stellar-cli: https://github.com/stellar/soroban-tools/pull/1156#discussion_r1446852810

I agree that moving this from the top level may make it easier to merge the CLIs, but I also understand that putting it under `contract` may make it feel more specific than the intended command. 🤔

Reply:
@leighmcculloch I'd like to see us keep this as a top-level command. As we increase support for classic stellar functionality in the CLI, I think the right thing to do with `init` would be to change the template so that it also has better support for classic Stellar. Nesting this under `contract` feels a little confusing to me, since it's not for initializing a contract. Are you ok with moving it back to the top level?

Reply:
I guess I'm not seeing what an init command for classic stellar would contain. Classic stellar applications are typically built using SDKs, building XDR txns, and calling the Horizon API. There's no common shape that I think we've ever found applications doing these things, although maybe we could generate a project that ingests data, or a project that is a frontend for using freighter.

Is that what you're thinking?

I think if we go that path we really need a flag that identifies a "type" of project the init function is generating. In the case of this PR, it adds a "contract" category.

Reply:
Ok well in that case I think keeping it where it is for now also makes sense.

Reply:
Thinking about this more, I think this could be quite useful especially as we encourage patterns for ingesting meta and building indexers. It could be helpful to init a project that is an indexer. So I think you're onto something and `soroban init --type contract` isn't too dissimilar of `soroban contract init`.

@chadoh @elizabethengelman I defer to you folks.

Reply:
Thanks for the good convo on this! Leaving it as `soroban contract init` for now seems good to me.

I like the idea of creating templates for indexers or ingesting metadata, but maybe we shouldn't include the `--type` flag yet, just in case we end up not building those additional templates. Also, I could see a world where `soroban indexer init` or `soroban project init` could be a thing. So maybe we can wait and see how it progresses to add the idea of different initializers.